<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>450</width>
    <height>450</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>450</width>
    <height>450</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>450</width>
    <height>450</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>431</width>
      <height>400</height>
     </rect>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::TabShape::Rounded</enum>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <property name="tabsClosable">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="tab_1">
     <attribute name="title">
      <string>AES Encrypt/Decrypt</string>
     </attribute>
     <attribute name="toolTip">
      <string>AES-based file encryption and decryption.</string>
     </attribute>
     <widget class="QTextEdit" name="textEdit_log">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>401</width>
        <height>171</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="placeholderText">
       <string>Status messages will appear here...</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>1</y>
        <width>401</width>
        <height>114</height>
       </rect>
      </property>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label_mode">
         <property name="text">
          <string>Mode:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="comboBox_mode">
         <item>
          <property name="text">
           <string>Encrypt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Decrypt</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_input">
         <property name="text">
          <string>Input File:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <layout class="QHBoxLayout" name="inputFileLayout">
         <item>
          <widget class="QLineEdit" name="lineEdit_input">
           <property name="acceptDrops">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_browseInput">
           <property name="text">
            <string>Browse...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_output">
         <property name="text">
          <string>Output File:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <layout class="QHBoxLayout" name="outputFileLayout">
         <item>
          <widget class="QLineEdit" name="lineEdit_output"/>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_browseOutput">
           <property name="text">
            <string>Browse...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_password">
         <property name="text">
          <string>Password:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="lineEdit_password">
         <property name="echoMode">
          <enum>QLineEdit::EchoMode::Password</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QPushButton" name="pushButton_start">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>401</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Start</string>
      </property>
     </widget>
     <widget class="QProgressBar" name="progressBar">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>340</y>
        <width>401</width>
        <height>23</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Hash Digest</string>
     </attribute>
     <attribute name="toolTip">
      <string>Compute the hash digest (e.g., MD5, SHA-256) of the input text using the chosen algorithm.</string>
     </attribute>
     <widget class="QComboBox" name="comboAlgorithm">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>291</width>
        <height>23</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Select the hash algorithm to use (e.g., SHA-256).</string>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="plainTextInput">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>291</width>
        <height>51</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Enter the text to be hashed.</string>
      </property>
      <property name="plainText">
       <string/>
      </property>
      <property name="placeholderText">
       <string>Enter the text to be hashed.</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnHashText">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>150</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Click to compute the hash of the text above.</string>
      </property>
      <property name="text">
       <string>Hash text</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnSelectFile">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>120</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Browse</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEditFilePath">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>211</width>
        <height>21</height>
       </rect>
      </property>
      <property name="acceptDrops">
       <bool>true</bool>
      </property>
      <property name="toolTip">
       <string>Path of the file to hash.</string>
      </property>
      <property name="placeholderText">
       <string>Drop a file here or click Browse.</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnHashFile">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>150</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::ContextMenuPolicy::NoContextMenu</enum>
      </property>
      <property name="toolTip">
       <string>Click to compute the hash of the selected file.</string>
      </property>
      <property name="text">
       <string>Hash file</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEditHashResult">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>180</y>
        <width>401</width>
        <height>151</height>
       </rect>
      </property>
      <property name="markdown">
       <string/>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Microsoft JhengHei UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="placeholderText">
       <string>The result of the hash will appear here.  </string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Algorithm:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_copyHash">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>150</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Copy Hash</string>
      </property>
     </widget>
     <widget class="QProgressBar" name="progressBar_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>340</y>
        <width>401</width>
        <height>23</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>GPG Decrypt</string>
     </attribute>
     <widget class="QProgressBar" name="progressBarGPG">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>340</y>
        <width>401</width>
        <height>23</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEditGPG_log">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>401</width>
        <height>171</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="placeholderText">
       <string>Status messages will appear here...</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_decryptGPG">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>401</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Decrypt</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>401</width>
        <height>71</height>
       </rect>
      </property>
      <layout class="QFormLayout" name="formLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label_input_2">
         <property name="text">
          <string>Input File:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <layout class="QHBoxLayout" name="inputFileLayout_2">
         <item>
          <widget class="QLineEdit" name="lineEditGPG_input">
           <property name="acceptDrops">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButtonGPG_browseInput">
           <property name="text">
            <string>Browse...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_output_2">
         <property name="text">
          <string>Output File:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <layout class="QHBoxLayout" name="outputFileLayout_2">
         <item>
          <widget class="QLineEdit" name="lineEditGPG_output"/>
         </item>
         <item>
          <widget class="QPushButton" name="pushButtonGPG_browseOutput">
           <property name="text">
            <string>Browse...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>450</width>
     <height>33</height>
    </rect>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::ContextMenuPolicy::DefaultContextMenu</enum>
   </property>
   <property name="acceptDrops">
    <bool>true</bool>
   </property>
   <property name="nativeMenuBar">
    <bool>true</bool>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
